core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -89,6 +93,88 @@
     public static final ParseField PARAMS_PARSE_FIELD = new ParseField("params");
 
     /**
+     * Constructs and validates a {@link Script} being parsed as XContent.
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
I wonder if this is actually easier to read than making a building with non-final fields and using that. If you think it is I'm happy to accept that but it might be worth writing it out the other and having a look just to make sure.